### PR TITLE
Easily add display headings with sass maps

### LINKED ISSUE #26515: More display-* headings - .display-5
### New .display-5 class

When creating website banners I often find myself using one of the `.display-*` headings from Bootstrap. They're clean, simple and look great!

The problem I've encountered is I find myself slightly restricted with the current `.display-*` classes, more specifically with the sizing, I often find myself needing something slightly smaller than `.display-4` and often tend to create my own `.display-5` class, and in the very rare occasion a `.display-6` class.

Could we extend this part and create an additional `.display-65` class?

**Something like:**

```
.display-5 {
    font-size: 2.5rem;
    font-weight: 300;
    line-height: 1.2;
}
```

I'm sure this wouldn't take too long to add?

**This is a mock-up of what the sizing would look like:**

display-5-class

Many Thanks!

:-)

Reference: https://getbootstrap.com/docs/4.1/content/typography/

### PR DESCRIPTION
**Pull request by MartijnCuppens**
Fixes #26515.

Provides a way to easily add display headings with a sass map.

### CODE REVIEW

**Comment by sanchez2210** [comment]
Will this be possible with normal text utilities? #25832

File: scss/_variables.scss [hunk 1]
```diff
@@ -260,16 +260,32 @@ $headings-font-weight:        500 !default;
...
-$display2-size:               5.5rem !default;
-$display3-size:               4.5rem !default;
-$display4-size:               3.5rem !default;
-
-$display1-weight:             300 !default;
-$display2-weight:             300 !default;
-$display3-weight:             300 !default;
-$display4-weight:             300 !default;
-$display-line-height:         $headings-line-height !default;
+$displays: (
```
Comment (ocarreterom) [inline scss/_variables.scss]:
```scss
$displays: () !default;
$displays: map-merge($displays, ());
```

**Comment by andresgalante** [comment]
Hi @MartijnCuppens, this solution would gives more settings and it works.

My problem is that by having the definition for headings scattered into different files, instead of scoped to type.scss we would be making the codebase harder to contribute, code and maintain.

What do you think?

**Comment by MartijnCuppens** [comment]
Hi @andresgalante,  

Would you move the `$displays` sass map to the `_type.scss`? All bootstrap variables are defined in `_variables.sccs`, I think it's better to keep it there?

**Comment by andresgalante** [comment]
Yes, you are right, all variables are define on variables and putting them within a map wouldn't make a difference than having them as variables outside the map.

**Review by mdo — DISMISSED** [review]
You've removed several variables—that's a breaking change. You can likely rewrite the Sass map to include those variables rather than remove them outright.

**Comment by MartijnCuppens** [comment]
I re-added the original display variables

**Comment by MartijnCuppens** [comment]
I'm not convinced anymore that this is a good idea (especially for v4). This technique is quite different from the way we handle things in other places, we're practically putting every property of the generated css in a sass map.

I'm considering closing this, unless someone has a good argument to merge this.

**Comment by XhmikosR** [comment]
CC @andresgalante @mdo

**Comment by mdo** [comment]
Let's punt on it and reimagine the typography in v5 to do it right. v5 shouldn't be sweeping changes, but an easy-ish upgrade path. I feel like stuff like this can be tackled then. Feel free to close if you still agree! <3

**Pull request closed without merging**